### PR TITLE
Refactor: rename image extraction kwargs

### PR DESCRIPTION
**Pull request by christinestraub**
Currently, we're using different kwarg names in partition() and partition_pdf(), which has implications for the API since it goes through partition().

### Summary
- rename `extract_element_types` -> `extract_image_block_types`
- rename `image_output_dir_path` to `extract_image_block_output_dir`
- rename `extract_to_payload` -> `extract_image_block_to_payload`
- rename `pdf_extract_images` -> `extract_images_in_pdf` in `partition.auto`
- add unit tests to test element extraction for `pdf/image` via `partition.auto`
### Testing
CI should pass.

### CODE REVIEW

**Comment by Coniferish** [comment]
Looks good. Only question/concern is if `extract_images_in_pdf` needs to just be deprecated instead of replaced. I'd assume if it was recently added it'd be safe to do so, but will leave that to @qued 's discretion.

**Comment by christinestraub** [comment]
> Looks good. Only question/concern is if `extract_images_in_pdf` needs to just be deprecated instead of replaced. I'd assume if it was recently added it'd be safe to do so, but will leave that to @qued 's discretion.

CC: @cragwolfe

File: unstructured/partition/pdf.py [hunk 1]
```diff
@@ -172,16 +172,16 @@ def partition_pdf(
         with Tesseract, you'll first need to install the appropriate Tesseract language pack.
     metadata_last_modified
         The last modified date for the document.
-    extract_images_in_pdf
```
Comment (cragwolfe) [inline unstructured/partition/pdf.py:175]: let's keep the old kwarg around as a deprecated argument for awhile to avoid breaking changes.

**Comment by Coniferish** [comment]
[re: comment here](https://github.com/Unstructured-IO/unstructured/pull/2310) I wish I had thought of this earlier, but with so many interdependent parameters, I wonder if it would made more sense to have `extract_to_payload` be a Tuple(str, list) instead of a list. Then the first index could be used to determine `extract_to_payload` or `image_output_dir_path` since those two parameters seem to be mutually exclusive.

**Comment by christinestraub** [comment]
> [re: comment here](https://github.com/Unstructured-IO/unstructured/pull/2310) I wish I had thought of this earlier, but with so many interdependent parameters, I wonder if it would made more sense to have `extract_to_payload` be a Tuple(str, list) instead of a list. Then the first index could be used to determine `extract_to_payload` or `image_output_dir_path` since those two parameters seem to be mutually exclusive.

@Coniferish Are you referring to something like this?
CC: @cragwolfe
```
element_extraction=("payload", ["Image", "Table"])
or 
element_extraction = ("output_dir", ["Image", "Table"])
```

**Comment by Coniferish** [comment]
> @Coniferish Are you referring to something like this?

Yah, that's what I was thinking @christinestraub

**Review by Coniferish — APPROVED** [review]
LGTM! I like these names a lot more!

**Review by Coniferish — CHANGES_REQUESTED** [review]
Just noticing that this is removing the old params, but @cragwolfe mentioned keeping/deprecating them instead

**Comment by christinestraub** [comment]
> Just noticing that this is removing the old params, but @cragwolfe mentioned keeping/deprecating them instead

For the `extract_images_in_pdf` param, I kept it in both `auto.partition()` and `partition.pdf()`
For the `image_output_dir_path` param, I renamed it to `extract_image_block_output_dir` since this param is optional (can be omitted), so this update will not bring the breaking change. CC: @cragwolfe

**Review by Coniferish — APPROVED** [review]

**Pull request merged**